src/controllers/queryRunner.ts

```diff
@@ -355,63 +356,84 @@ export default class QueryRunner {
     public async copyResults(selection: ISlickRange[], batchId: number, resultId: number, includeHeaders?: boolean): Promise<void> {
         let copyString = '';
 
-        // if just copy headers
-        if (!selection && this.shouldIncludeHeaders(includeHeaders)) {
-            const columns = this.batchSets[batchId].resultSetSummaries[resultId].columnInfo.length;
-            const headerRange: ISlickRange = {
-                fromRow: 0,
-                fromCell: 0,
-                toRow: 0,
-                toCell: columns
-            };
-            let columnHeaders = this.getColumnHeaders(batchId, resultId, headerRange);
-            if (columnHeaders !== undefined) {
-                copyString += columnHeaders.join('\t') + os.EOL;
-            }
-            await this._vscodeWrapper.clipboardWriteText(copyString);
-            let oldLang: string;
-            if (process.platform === 'darwin') {
-                oldLang = process.env['LANG'];
-                process.env['LANG'] = 'en_US.UTF-8';
-            }
-            await this._vscodeWrapper.clipboardWriteText(copyString);
-            if (process.platform === 'darwin') {
-                process.env['LANG'] = oldLang;
+        // add the column headers
+        if (this.shouldIncludeHeaders(includeHeaders)) {
+            let firstCol: number;
+            let lastCol: number;
+            for (let range of selection) {
+                if (firstCol === undefined || (range.fromCell < firstCol)) {
+                    firstCol = range.fromCell;
+                }
+                if (lastCol === undefined || (range.toCell > lastCol)) {
+                    lastCol = range.toCell;
+                }
             }
+            let columnRange: ISlickRange = {
+                fromCell: firstCol,
+                toCell: lastCol,
+                fromRow: undefined,
+                toRow: undefined
+            };
+            let columnHeaders = this.getColumnHeaders(batchId, resultId, columnRange);
+            copyString += columnHeaders.join('\t');
         }
+        copyString += os.EOL;
```

Comment:
Oops, one more thing. So just noticed that we actually check if headers should be included. So two things :

1. We shouldn't always be adding on the EOL in that case (having an extra line above copied content would be annoying)
2. In that case we need to check at the end that copystring.length > 1 before trimming off the last EOL since it'd be possible for the string to be empty